### ISSUE TITLE
recently updated google apps break

### ISSUE THREAD

**Issue by Vendicated** [microg/GmsCore#1373]
**Affected apps**
Name: YouTube v16+
Package id: com.google.android.youtube

Name: Youtube Music v4.10+
Package id: com.google.android.apps.youtube.music

**Describe the bug**

As described in the title, Google apps that were recently updated break. YouTube loads indefinitely, YouTube Music simply does not let you login. 

**To Reproduce**

- Install YouTube/Youtube Music (or probably any other newly updated Google app) 

**Expected behavior**

Apps work regularly

**Screenshots**
![image](https://user-images.githubusercontent.com/45497981/105611569-fe2cb400-5dad-11eb-9006-2779930ecf65.png)

**System**
Android Version: 10
Custom ROM: Havoc-OS 3.12

**microG**
microG Core version: 0.2.16.204713 (I updated to v0.2.17.204714 after realising i wasnt on latest, still broken)
microG Self-Check results: All ticked

**Additional context**

```yaml
01-23 19:45:49.984 10608 10630 W GooglePlayServicesUtil: Google Play services out of date for com.google.android.youtube.  Requires 210290000 but found 204713032
```

Hardcoding the version to the required one doesnt fix it, so there seems to be more to it than just bumping the version number

<details> 
  <summary>There is also this logcat output which might also be related to it:</summary>

```yaml
01-23 19:45:51.338 10608 10635 W Auth    : [GoogleAuthUtil] Service call returned null.
01-23 19:45:51.389 10608 10629 W Auth    : [GoogleAuthUtil] Service call returned null.
01-23 19:45:51.392 10608 10629 W Auth    : [GoogleAuthUtil] Service call returned null.
01-23 19:45:51.406 10608 10608 W View    : requestLayout() improperly called by android.widget.FrameLayout{bbcf7f V.E...... ......I. 0,126-1080,1857 #7f0b0a50 app:id/pane_fragment_container} during layout:
running second layout pass
01-23 19:45:51.536 10608 10658 W Auth    : [GoogleAuthUtil] Service call returned null.
01-23 19:45:51.539 10608 10658 W Auth    : [GoogleAuthUtil] Service call returned null.
```
</details>

<details> 
  <summary>And some more</summary>

```yaml
01-23 19:55:45.063 10608 10608 W ProtoDataStoreFlagStore: Unable to retrieve flag snapshot for com.google.android.libraries.performance.primes#com.google.android.youtube, using defaults.
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore: Unable to update local snapshot for com.google.android.libraries.performance.primes#com.google.android.youtube, may result in stale flags.
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore: java.util.concurrent.ExecutionException: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass
()' on a null object reference
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at athg.s(PG:6)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at athg.get(PG:4)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at atlf.b(PG:1)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at atki.u(PG:2)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at yjz.c(Unknown Source:0)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at yjx.run(Unknown Source:4)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at java.lang.Thread.run(Thread.java:919)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at yho.a(Unknown Source:6)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at vos.run(PG:2)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at atio.execute(PG:1)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at vot.a(PG:1)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at vpp.b(PG:5)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at vpu.p(PG:3)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at vpq.a(PG:1)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at ugm.b(PG:2)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at vnn.e(PG:1)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at vnp.dispatchTransaction(PG:34)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at dlg.onTransact(PG:2)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at android.os.Binder.execTransactInternal(Binder.java:1021)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at android.os.Binder.execTransact(Binder.java:994)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at android.os.BinderProxy.transactNative(Native Method)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at android.os.BinderProxy.transact(BinderProxy.java:511)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at dlf.nT(PG:2)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at vnr.f(PG:6)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at vnk.a(Unknown Source:3)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at ugi.a(PG:1)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at ucq.f(PG:1)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at ueu.r(PG:2)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at ueu.q(PG:5)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at ueu.g(PG:3)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at ueu.d(PG:13)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at ueu.oE(PG:2)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at uib.a(PG:1)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at uhi.a(PG:9)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at uha.d(Unknown Source:6)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at uhd.handleMessage(PG:37)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at android.os.Handler.dispatchMessage(Handler.java:107)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at android.os.Looper.loop(Looper.java:214)
01-23 19:55:45.176 10608 10913 W ProtoDataStoreFlagStore:       at android.os.HandlerThread.run(HandlerThread.java:67)
```

</details>

**Comment by KevinX8** [comment]
I can confirm the issue is not related to version code, bumping it does not fix the issue, also if you update from for example 15.50 or earlier to 16.X then the infinite loading issue will not occur.

**Comment by Oizaro** [comment]
@mar-v-in This is a really terrible bug. :/

**Comment by mastercraft01** [comment]
![Uploading Screenshot_2021-01-28-04-47-12-98.png…]()

**Comment by Osahashi** [comment]
It works without problem on my LG V30, but my Huawei Mediapad 5 has the same issue, I tried everything but the its not working. Its a new problem with version 16 now... The version 15 works on my Huawei Mediapad.

**Comment by wefalltomorrow** [comment]
There's a temporary 'solution' for this until the core problem is fixed. https://www.reddit.com/r/Vanced/comments/l6ehkn/other_fix_vanced_160235/

**Comment by DarthCluster** [comment]
Cannot pause video while casting...

**Comment by gghhkm** [comment]
#1388

**Comment by Dystination** [comment]
Work oround: 

Install an older Version of Youtube Vanced.
I also installed an older version of MicroG but i dont know if this was necessary, bud i did it anyways.

https://www.apkmirror.com/apk/team-vanced/youtube-vanced/youtube-vanced-15-43-32-release/youtube-vanced-15-43-32-android-apk-download/download/

**Comment by rohittp0** [comment]
> Workaround:
> 
> Install an older Version of Youtube Vanced.
> I also installed an older version of MicroG but i dont know if this was necessary, bud i did it anyways.
> 
> https://www.apkmirror.com/apk/team-vanced/youtube-vanced/youtube-vanced-15-43-32-release/youtube-vanced-15-43-32-android-apk-download/download/

I tried this method but it is doesn't allow to swich account. This can be a real bummer for those who have more than 1 account.

**Comment by andres8012** [comment]
It only served me for one account, the others no longer seem to me😔

**Comment by Rihcus** [comment]
One alternate fix (oneplus only idk if other phones have this) is to use parallel apps. You can then install Vanced manager to user 999 and install micro g and Vanced (micro g can be uninstalled as if theain app is updated the parallel app is auto updated).

**Comment by Rihcus** [comment]
Alternatively if Vanced 15 is resigned with a different signature it may be possible to run 15 and 16 side by side (and maybe use same micro g not sure though)

**Comment by KevinX8** [comment]
Team Vanced has now opened a €100 l worth of Basic Attention Token bounty for anyone who manages to resolve this issue as it has been open for close to two months now.

**Comment by AboAzzh** [comment]
Don't download or update the app to the last version just use the older version 
Keep trying in the older version until it's work in your phone

**Comment by Vendicated** [comment]
I'm not sure why you guys think this is the right place to get help with Vanced related things. This is the upstream microg repository and has nothing to do with Vanced. Please stop turning this into some sort of support thread and stay on topic 


Le 12 mars 2021 06:54:24 UTC, AboAzzh ***@***.***> a écrit :

**Comment by rohittp0** [comment]
> I'm not sure why you guys think this is the right place to get help with Vanced related things. This is the upstream microg repository and has nothing to do with Vanced. Please stop turning this into some sort of support thread and stay on topic Le 12 mars 2021 06:54:24 UTC, AboAzzh ***@***.***> a écrit :
> […](#)
> Don't download or update the app to the last version just use the older version Keep trying in the older version until it's work in your phone -- You are receiving this because you authored the thread. Reply to this email directly or view it on GitHub: [#1373 (comment)](https://github.com/microg/GmsCore/issues/1373#issuecomment-797278758)

This is because the issue is not with vanced but with microg.

**Comment by X0Predator** [comment]
Yeah , I already have faced this like 4 times in the past 2 weeks , I don't know but the wrong code part , the solution is uninstall everything (Vanced+microG) (microG also affected) , install microG first , install Vanced v15 , open both , wait to load , update to v16 and carry on .

**Comment by X0Predator** [comment]
> Yeah , I already have faced this like 4 times in the past 2 weeks , I don't know but the wrong code part , the solution is uninstall everything (Vanced+microG) (microG also affected) , install microG first , install Vanced v15 , open both , wait to load , update to v16 and carry on .

also Normal YT version *must be same/less than* Vanced

**Comment by BlackXS** [comment]
I am on non-rooted huawei mate 20 .. emui 11
Microg 0.2.16.204713
And on latest version of vanced music (4.11.50-b2.4) and YouTube vanced (16+)
And everything works well and the youtube vanced stays and it doesn't change back to stock and yt vanced music as well

**Comment by gghhkm** [comment]
@BlackXS
@X0Predator
Can you guys plz stop talking here, this is ***not*** for Vanced support.
Plz don't talk here unless you are giving a way to patch this particular bug in MicroG from MicroG Project.

**Comment by 98mohitkumar** [comment]
[installed google play services 21.06.13 (120400-358943053)]
(https://www.apkmirror.com/apk/google-inc/google-play-services/google-play-services-21-06-13-release/google-play-services-21-06-13-120400-358943053-android-apk-download/)

then updated youtube vanced from 15.43.32 to 16.02.35 and it worked just fine

currently installed micro G version :  0.2.16.204713

**Comment by Vendicated** [comment]
Fixed in Youtube 16.08 and Music 4.18.

**Comment by gghhkm** [comment]
> Fixed in Youtube 16.08 and Music 4.18.

Nice.
